src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -109,48 +106,167 @@ class boss_gluth : public CreatureScript
+
+                        // the soon-to-be-eaten zombie should stop moving and stop attacking
+                        zombie->StopMoving();
+                        zombie->GetMotionMaster()->Clear();
+                        zombie->GetMotionMaster()->MoveIdle();
+
+                        zombie->SetTarget(ObjectGuid::Empty);
+                        zombie->getThreatManager().clearReferences();
+                        zombie->CombatStop();
+                        zombie->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
+                        zombie->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                        zombie->SetReactState(ReactStates::REACT_PASSIVE);
+
+                        // gluth should turn toward the zombie and stop AAs on his primary target. He then pauses for a few seconds
+                        me->GetMotionMaster()->MoveFollow(zombie, 0, 0);
+                        me->SetReactState(ReactStates::REACT_PASSIVE);
+                        me->SetFacingToObject(zombie);
+                        me->SetTarget(ObjectGuid::Empty);
+
+                        events.ScheduleEvent(EVENT_KILL_ZOMBIE, 2.5 SECONDS);
+                    }
+
+                    events.ScheduleEvent(EVENT_SEARCH_ZOMBIE, 7 SECONDS);
+                    break;
+                case EVENT_KILL_ZOMBIE:
+                    if (!zombie || !zombie->IsAlive()) // some of the tests are a bit overkill but better be safe than sorry, right?
                         break;
+
+                    // do the actual killing
+                    DoCast(zombie, SPELL_ZOMBIE_CHOW_SEARCH);
+                    zombie = nullptr;
+
+                    // and then return on primary target
+                    me->GetMotionMaster()->MoveChase(me->GetVictim());
+                    me->SetFacingToObject(me->GetVictim());
+                    me->SetTarget(me->GetVictim()->GetGUID());
+                    me->SetReactState(ReactStates::REACT_AGGRESSIVE);
+
+                    // also, heal Gluth for 5% of his hps
+                    me->ModifyHealth(int32(me->CountPctFromMaxHealth(5))); //heal gluth for 5% of his hps
+                    break;
                 }
             }
 
-            if (me->GetVictim() && me->EnsureVictim()->GetEntry() == NPC_ZOMBIE)
+            if (me->HasReactState(REACT_AGGRESSIVE))
+                DoMeleeAttackIfReady();
+        }
+
+        void DecimateEvent()
+        {
+            for (SummonList::const_iterator itr = summons.begin(); itr != summons.end(); ++itr)
+            {
+                zombie = ObjectAccessor::GetCreature(*me, *itr);
+                zombie->SetReactState(ReactStates::REACT_PASSIVE);
+                zombie->GetMotionMaster()->MoveFollow(me, 0, 0);
+                zombie->AttackStop();
+                zombie->SetTarget(ObjectGuid::Empty);
+
+            }
+        }
+
+    private:
+        Creature* zombie;
```

Comment:
It has caused many issues in the past due to dangling references, thread races and so on.
Also, it seems that most of your uses of `Creature* zombie` can and should be replaced by local variables.